adapters/Vungle/VungleAdapter/VungleRouter.m

```diff
@@ -127,22 +246,82 @@ - (BOOL)playAd:(UIViewController *)viewController
                              options:options
                          placementID:delegate.desiredPlacement
                                error:&error]) {
-    startPlaying = false;
+    didAdStartPlaying = false;
   }
   if (error) {
     NSLog(@"Adapter failed to present ad, error %@", [error localizedDescription]);
-    startPlaying = false;
-  };
-  return startPlaying;
+    didAdStartPlaying = false;
+  }
+
+  return didAdStartPlaying;
+}
+
+- (UIView *)renderBannerAdInView:(UIView *)bannerView
+                        delegate:(id<VungleDelegate>)delegate
+                          extras:(VungleAdNetworkExtras *)extras
+                  forPlacementID:(NSString *)placementID {
+  NSError *bannerError = nil;
+  NSMutableDictionary *options = [[NSMutableDictionary alloc] init];  ///
+  if (extras.muted) {
+    [VungleSDK sharedSDK].muted = extras.muted;
+  } else {
+    [VungleSDK sharedSDK].muted = YES;
+  }
+  if (extras.userId) [options setObject:extras.userId forKey:VunglePlayAdOptionKeyUser];
+  if (extras.ordinal) [options setObject:@(extras.ordinal) forKey:VunglePlayAdOptionKeyOrdinal];
+  if (extras.flexViewAutoDismissSeconds)
+    [options setObject:@(extras.flexViewAutoDismissSeconds)
+                forKey:VunglePlayAdOptionKeyFlexViewAutoDismissSeconds];
+
+  BOOL success = [[VungleSDK sharedSDK] addAdViewToView:bannerView
+                                            withOptions:options
+                                            placementID:placementID
+                                                  error:&bannerError];
+  if (success) {
+    if (delegate.adapterAdType == MREC) {
+        self.isMrecPlaying = YES;
+    } else {
+        self.isBannerPlaying = YES;
+    }
+
+    return bannerView;
+  } else {
+    NSLog(@"Banner loading error: %@", bannerError.localizedDescription);
+    if (delegate.adapterAdType == MREC) {
+        self.isMrecPlaying = NO;
+    } else {
+        self.isBannerPlaying = NO;
+    }
+
+    return nil;
```

Comment:
since the method will return `nil` is this needed?

Reply:
Got it, this “return nil” is useless and I will remove it.